### PR TITLE
Format multiple extraneous fields in a human readable manner

### PR DESCRIPTION
**Pull request by nyergler**
https://lobsters.atlassian.net/browse/ENG-6792

**What:**

Introduce the idea of language templates that can be used for formatting errors coming from Joi; provide a default template that formats extraneous fields to be more readable by humans.

Before: `"foo is not allowed or foo2 is not allowed or foo3 is not allowed"`

After: `the following parameters are not allowed: foo, foo2, foo3`

**Why:**

Introducing language templates allows us -- and other consumers of the package -- to configure how hapi-format-error processes our error messages.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -29,17 +43,56 @@ exports.register = function (server, options, next) {
...
+        }, {}
+      );
+
+      // process each group, then flatten the final array of arrays
+      msg = Hoek.flatten(
+        Object.keys(details).map(
+          (err_type) => {
+            const template = Hoek.reach(options.language, err_type);
+
+            if (Array.isArray(template)) {
```
[the remark this answers is not in the document]
Comment (nyergler) [reply lib/index.js]: Yeah, good idea; I also think that could streamline the logic overall.

**Comment by nyergler** [comment]
@pallavi Updated to use an object rather than an array for templates based on your suggestion.

File: test/index.test.js [hunk 2]
```diff
@@ -305,4 +314,66 @@ describe('format error plugin', () => {
...
+    return server.injectThen({
+      method: 'POST',
+      url: '/no-params',
+      payload: {
+        foo: true,
+        bar: false
+      }
+    })
+    .then((res) => {
+      expect(res.result.error.message).to.eql('the following parameters are not allowed: foo,bar');
```
Comment (mgartner) [inline test/index.test.js]: In your PR description, the example error messages include a space after the commas, which I think makes the error message more humanly readable.

Do you think the default template should format plurals like this with a space after the `,`? Or do you see that being something a user would specify in their template with something like:

```node
  plural: 'the following parameters are not allowed: {paths.join(', ')}'
```

Which brings up another question, is executable code (like `paths.join()`) allowed in templated?

Comment (nyergler) [reply test/index.test.js]: Yeah, the spaces are more humane. The templates follow the same rules as Hapi language templates, so they do not allow executable expressions.

I'll update this to makes `paths` a string (possible renamed to `paths_str`) joined by `', '` before merging.

**Pull request merged**